core/types/calls.cc

```diff
@@ -92,12 +93,10 @@ DispatchResult AndType::dispatchCall(const GlobalState &gs, const DispatchArgs &
         leftRet = left.dispatchCall(gs, args.withThisRef(left));
         rightRet = right.dispatchCall(gs, args.withThisRef(right));
     }
-    DispatchResult ret{Types::all(gs, leftRet.returnType, rightRet.returnType), move(leftRet.main),
-                       make_unique<DispatchResult>(move(rightRet)),
 
-                       DispatchResult::Combinator::AND};
-
-    return ret;
+    auto resultType = Types::all(gs, leftRet.returnType, rightRet.returnType);
+    return DispatchResult::merge(std::move(resultType), DispatchResult::Combinator::AND, std::move(leftRet),
+                                 std::move(rightRet));
```

Comment:
This is super-minor, and probably not worth it given that `DispatchResult::merge` is only liable to get used in two places, but WDYT about making `merge` instead just take the combinator type and the left/right results?  Then you can't get the result type wrong.

Reply:
The result type is different for each case: it's `Types::any` for the OrType case and `Types::all` for the `AndType` case.

Reply:
Or were you suggesting casing on the combinator for figuring out which type to make?